controller/service.go

```diff
@@ -114,12 +115,15 @@ func (c *Controller) getPreviewAndActiveServices(r *v1alpha1.Rollout) (*corev1.S
 	var previewSvc *corev1.Service
 	var activeSvc *corev1.Service
 	var err error
-	logCtx := logutil.WithRollout(r)
 	if r.Spec.Strategy.BlueGreenStrategy.PreviewService != "" {
 		previewSvc, err = c.kubeclientset.CoreV1().Services(r.Namespace).Get(r.Spec.Strategy.BlueGreenStrategy.PreviewService, metav1.GetOptions{})
 		if err != nil {
 			if errors.IsNotFound(err) {
-				logCtx.Warnf("Service %v does not exist", r.Spec.Strategy.BlueGreenStrategy.PreviewService)
+				msg := fmt.Sprintf(conditions.ServiceNotFoundMessage, "preview", r.Spec.Strategy.BlueGreenStrategy.PreviewService)
+				c.recorder.Eventf(r, corev1.EventTypeWarning, conditions.ServiceNotFoundReason, msg)
+				cond := conditions.NewRolloutCondition(v1alpha1.RolloutProgressing, corev1.ConditionFalse, conditions.ServiceNotFoundReason, msg)
+				conditions.SetRolloutCondition(&r.Status, *cond)
+				c.rolloutsclientset.ArgoprojV1alpha1().Rollouts(r.Namespace).Update(r)
```

Comment:
I think we should use Patch to set conditions.

Reply:
👍